apps/web/app/bookings/[status]/page.tsx

```diff
@@ -0,0 +1,29 @@
+import { PageProps } from "app/_types";
+import { _generateMetadata } from "app/_utils";
+import { WithLayout } from "app/layoutHOC";
+import { redirect } from "next/navigation";
+import { z } from "zod";
+
+import { validStatuses } from "~/bookings/lib/validStatuses";
+import BookingsList from "~/bookings/views/bookings-listing-view";
+
+const querySchema = z.object({
+  status: z.enum(validStatuses),
+});
+
+export const generateMetadata = async () =>
+  await _generateMetadata(
+    (t) => t("bookings"),
+    (t) => t("bookings_description")
+  );
+
+const Page = async ({ params, searchParams }: PageProps) => {
+  const parsed = querySchema.safeParse({ ...params, ...searchParams });
+  if (!parsed.success) {
+    redirect("/bookings/upcoming");
```

Comment:
- In Production, we currently take user to 404 if they go to URL like `/bookings/123` because 123 is not a valid status (upcoming, cancelled, etc).
- With this change, we redirect user to /bookings/upcoming instead

Reply:
Now that we don't have `dynamic = "force-static"`, does this mean this page is server rendered every time? Or Next.js still builds it statically because of `generateStaticParams`? I'd appreciate your explaination as I'm not familiar enough with these to understand the behavior 👀

Reply:
> Or Next.js still builds it statically because of generateStaticParams?

This is absolutely correct. With `dynamic = "force-static"`, user will get routed to 404 if `status` is not one of those returned by `generateStaticParams` before we have chance to redirect them to `/bookings/upcoming`.